source/grid/grid_in.cc

```diff
@@ -178,6 +178,9 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
     for (unsigned int i = 0; i < 4; ++i)
       {
         getline(in, line);
+
+        if (i == 0)
+          text[0] = line;
```

Comment:
Since you're overwriting `text[0]`, perhaps just don't initialize it above  (or leave a comment about what it will contain, say 
```
  // text[0] will contain the version string after reading the preamble.
```